test/test_utils_log.cpp

```diff
@@ -0,0 +1,380 @@
+    for (auto o : outs) {
+        loggerConfig = {0, 0, LOG_DEBUG, LOG_DEBUG, o};
+        expected_stream = o;
+        helper_test_log(LOG_DEBUG, "example log");
+    }
+}
+
+TEST_F(test, flush_levels) {
+    expected_stream = stderr;
+    expect_fput_count = 1;
+    for (int i = LOG_DEBUG; i <= LOG_ERROR; i++) {
+        for (int j = LOG_DEBUG; j <= LOG_ERROR; j++) {
+            loggerConfig = {0, 0, LOG_DEBUG, (utils_log_level_t)i, stderr};
+            if (i > j) {
+                expect_fflush_count = 0;
+            } else {
+                expect_fflush_count = 1;
+            }
+            expected_message = "[" + helper_log_str(j) + " UMF] example log\n";
+            helper_test_log((utils_log_level_t)j, "example log");
+        }
+    }
+}
+
+TEST_F(test, long_log) {
+    expect_fput_count = 1;
+    expect_fflush_count = 1;
+    loggerConfig = {0, 0, LOG_DEBUG, LOG_DEBUG, stderr};
+    expected_message = "[DEBUG UMF] " + std::string(8191, 'x') + "\n";
+    helper_test_log(LOG_DEBUG, std::string(8191, 'x').c_str());
+    expected_message =
+        "[DEBUG UMF] " + std::string(8191, 'x') + "[truncated...]\n";
+    helper_test_log(LOG_DEBUG, +std::string(8192, 'x').c_str());
+}
+
+TEST_F(test, timestamp_log) {
+    expect_fput_count = 1;
+    expect_fflush_count = 1;
+    loggerConfig = {1, 0, LOG_DEBUG, LOG_DEBUG, stderr};
+    // TODO: for now we do not check output message,
+    // as it requires more sophisticated message validation (a.k.a regrex)
+    expected_message = "";
+    helper_test_log(LOG_DEBUG, "example log");
+}
+
+TEST_F(test, pid_log) {
+    expect_fput_count = 1;
+    expect_fflush_count = 1;
+    loggerConfig = {0, 1, LOG_DEBUG, LOG_DEBUG, stderr};
+    // TODO: for now we do not check output message,
+    // as it requires more sophisticated message validation (a.k.a regrex)
+    expected_message = "";
+    helper_test_log(LOG_DEBUG, "example log");
+}
+
+TEST_F(test, log_macros) {
+    expected_stream = stderr;
+    expect_fput_count = 1;
+    expect_fflush_count = 1;
+    loggerConfig = {0, 0, LOG_DEBUG, LOG_DEBUG, stderr};
+
+    expected_message = "[DEBUG UMF] example log\n";
+    fput_count = 0;
+    fflush_count = 0;
+    DEBUG("example log");
+    EXPECT_EQ(fput_count, expect_fput_count);
+    EXPECT_EQ(fflush_count, expect_fflush_count);
+
+    expected_message = "[INFO  UMF] example log\n";
+    fput_count = 0;
+    fflush_count = 0;
+    INFO("example log");
+    EXPECT_EQ(fput_count, expect_fput_count);
+    EXPECT_EQ(fflush_count, expect_fflush_count);
+
+    expected_message = "[WARN  UMF] example log\n";
+    fput_count = 0;
+    fflush_count = 0;
+    WARN("example log");
+    EXPECT_EQ(fput_count, expect_fput_count);
+    EXPECT_EQ(fflush_count, expect_fflush_count);
+
+    expected_message = "[ERROR UMF] example log\n";
+    fput_count = 0;
+    fflush_count = 0;
+    ERR("example log");
+    EXPECT_EQ(fput_count, expect_fput_count);
+    EXPECT_EQ(fflush_count, expect_fflush_count);
+}
```

Comment:
can you pls add a test with a little longer logging, e.g.: log, log, log, flush, log, log, ... (especially with various log and flush levels)

Reply:
This is whitebox unit test  - we know that logger is stateless, so there is no need to test sequences.